string.go

```diff
@@ -57,17 +60,16 @@ func AESDecrypt(key, nonce, ciphertext []byte) ([]byte, error) {
 		return nil, err
 	}
 
-	plainText, err := gcm.Open(nil, nonce, ciphertext, nil)
-	if err != nil {
-		return nil, err
-	}
+	size := gcm.NonceSize()
+	nonce := ciphertext[:size]
```

Comment:
Done. Ideally there shouldn't be any issues with the length of the ciphertext since `AESGCMEncrypt` will fail and throw an error if it can't encrypt. Then there'd be no data to pass to `AESGCMDecrypt`, no?